### PR TITLE
Oxedions/fix/core simplification

### PR DESCRIPTION
**Pull request by oxedions**
Simplify core.

Repositories and DNS server moved to advanced, and should be able to receive more features in future.
New basic roles for core (without many features, be careful when using).

### CODE REVIEW

**Comment by oxedions** [comment]
Seems good. I will squash once validated before merging.

**Comment by oxedions** [comment]
Advanced roles may need a little more documentation in the readme files. Can be done later.

**Comment by actatux** [comment]
Hello,

Before reviewing, why did you change your mind about these 2 roles? We agreed to let these in core in [this thread](https://github.com/bluebanquise/bluebanquise/issues/182#issuecomment-640064232).

**Comment by oxedions** [comment]
I remember :-)

There are 2 reasons:

1 - I had some issues with the advanced roles: one was not compatible with the diskless addon, the second one was not compatible with the HA that my company is implementing. Both issues could simply be bypassed for an advanced system administrator, but I want core to work "as is", while I understand advanced core need advanced knowledge.
2 - Some users reported still too much complexity on these roles, which was not compatible with the didactic goal of the stack.

So I choose to move them into advanced, and create very basic ones for the very basic usage. The core ones will keep very basic or naive, while advanced ones will be able to offer more and more features as needed in the future and will concentrate developments focus. 

Same happen for dhcp server before.

**Comment by actatux** [comment]
1 - I would be pleased to have further details since we are working for the same company and I'm not aware of these issues.

2 - I will review then, but I would tend for very basic and simple roles. The current proposal is too complicated and, at first read, too close to the advanced roles.

File: roles/core/dns_server/tasks/main.yml [hunk 1]
```diff
@@ -42,61 +42,93 @@
...
+    group: root
+    mode: 0644
+  register: zoneconf
   tags:
-    - internal
+    - template
 
-# If dns_ip is a list, pick the first value as master DNS
-- name: set_fact ░ Set master DNS server - Pick first element of dns_ip
+- name: set_fact ░ Set SOA Serial for forward zone
```
Comment (actatux) [inline roles/core/dns_server/tasks/main.yml:81]: redundant with previous task line 62.

File: roles/core/dns_server/templates/named.conf.j2 [hunk 2]
```diff
@@ -74,43 +64,17 @@ zone "." IN {
   file "named.ca";
 };
 
-{# Is the host the master DNS? #}
-{% if dns_master in inventory_all_ipv4_addresses %}
```
Comment (actatux) [inline roles/core/dns_server/templates/named.conf.j2:78]: Beware, users might end with several master dns if they run the role on two management nodes. You need to ensure there is a single dns server for all the cluster.

File: roles/core/dns_server/vars/main.yml [hunk 3]
```diff
@@ -1,2 +1,2 @@
 ---
-dns_server_role_version: 1.1.1
+dns_server_role_version: 1.1.2
```
Comment (actatux) [inline roles/core/dns_server/vars/main.yml:2]: This is a new role, should be 1.0.0.

**Comment by johnnykeats** [comment]
Hello Ox.
As promised, will have a look. ;)

**Comment by actatux** [comment]
> 1 - I had some issues with the advanced roles: one was not compatible with the diskless addon,

I found out this is repositories_client. I opened #283 which should fix the issue in the role diskless.

**Comment by johnnykeats** [comment]
@oxedions 

Ok, got it now.
You are again trying to adapt the stack to your company needs. You accepted many things you were against of because of external pressures, which makes us unhappy.
1.2 version was stable, now it is a mess.

BlueBanquise is an **independent** stack. We had excellent feedback with the original philosophy and architecture.

Please revert and remove all the things you do not agree with before going further, then only will I review.

**Comment by oxedions** [comment]
So be it. I was as always way too kind.

I should have done this long time ago.

May take few days.

**Pull request closed without merging**